addons/l10n_ro_edi_stock/models/stock_picking.py

```diff
@@ -0,0 +1,864 @@
+        for picking in self:
+            picking.l10n_ro_edi_stock_available_start_loc_types = picking._l10n_ro_edi_stock_get_available_location_types('start')
+            picking.l10n_ro_edi_stock_available_end_loc_types = picking._l10n_ro_edi_stock_get_available_location_types('end')
+
+    @api.depends('l10n_ro_edi_stock_document_ids')
+    def _compute_l10n_ro_edi_stock_current_document_data(self):
+        for picking in self:
+            document = picking._l10n_ro_edi_stock_get_current_document()
+
+            picking.l10n_ro_edi_stock_state = document.state if document else False
+            picking.l10n_ro_edi_stock_document_message = document.message if document else False
+            picking.l10n_ro_edi_stock_document_load_id = document.l10n_ro_edi_stock_load_id if document else False
+            picking.l10n_ro_edi_stock_document_uit = document.l10n_ro_edi_stock_uit if document else False
+
+    @api.depends('company_id.country_code')
+    def _compute_l10n_ro_edi_stock_enable(self):
+        for picking in self:
+            picking.l10n_ro_edi_stock_enable = picking.company_id.country_code == 'RO'
+
+    @api.depends('l10n_ro_edi_stock_enable', 'state', 'l10n_ro_edi_stock_state')
+    def _compute_l10n_ro_edi_stock_enable_send(self):
+        for picking in self:
+            picking.l10n_ro_edi_stock_enable_send = (picking.l10n_ro_edi_stock_enable
+                                                     and picking.state == 'done'
+                                                     and picking.l10n_ro_edi_stock_state in (False, 'stock_sending_failed')
+                                                     and not picking._l10n_ro_edi_stock_get_last_document('stock_validated'))
+
+    @api.depends('company_id', 'state', 'l10n_ro_edi_stock_state')
+    def _compute_l10n_ro_edi_stock_enable_fetch(self):
+        for picking in self:
+            picking.l10n_ro_edi_stock_enable_fetch = picking.l10n_ro_edi_stock_enable and picking.l10n_ro_edi_stock_state == 'stock_sent'
+
+    @api.depends('l10n_ro_edi_stock_document_message')
+    def _compute_l10n_ro_edi_stock_enable_send_info(self):
+        for picking in self:
+            picking.l10n_ro_edi_stock_enable_send_info = bool(picking.l10n_ro_edi_stock_document_message)
+
+    @api.depends('l10n_ro_edi_stock_state')
+    def _compute_l10n_ro_edi_stock_enable_amend(self):
+        for picking in self:
+            picking.l10n_ro_edi_stock_enable_amend = picking.l10n_ro_edi_stock_enable and (
+                    picking.l10n_ro_edi_stock_state == 'stock_validated'
+                    or (picking.l10n_ro_edi_stock_state == 'stock_sending_failed'
+                        and picking._l10n_ro_edi_stock_get_last_document('stock_validated')))
+
+    @api.depends('l10n_ro_edi_stock_state')
+    def _compute_l10n_ro_edi_stock_fields_readonly(self):
+        for picking in self:
+            picking.l10n_ro_edi_stock_fields_readonly = picking.l10n_ro_edi_stock_state == 'stock_sent'
```

Comment:
you always check `l10n_ro_edi_stock_enable` but not here :thinking:

Reply:
we could add it but it is kind of redundant since having `picking.l10n_ro_edi_stock_state == 'stock_sent'` implies that `l10n_ro_edi_stock_enable` is true